tiledb/sm/cpp_api/query.h

```diff
@@ -511,7 +511,7 @@ class Query {
    */
   template <class T>
   std::array<T, 3> range(unsigned dim_idx, uint64_t range_idx) {
-    impl::type_check<T>(schema_.domain().type());
+    impl::type_check<T>(schema_.domain().dimension(dim_idx).type());
```

Comment:
note: I included this as I saw it when I was adding the new function, and realized it was missed the previous refactors.